src/restconfig/src/main/java/org/geoserver/rest/resources/ResourceResource.java

```diff
@@ -0,0 +1,409 @@
+        this.resource = resource;
+        this.operation = operation;
+        this.store = store;
+    }
+    
+    public Resource getResource() {
+        return resource;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+        
+    private String href(String path, boolean isDir) {
+        String href = getPageInfo().rootURI("resources/" + path);
+        if (isDir) {
+            String format = (String) getRequest().getAttributes().get("format");
+            if (format != null) {
+                href += "?format=" + format;
+            }
+        }
+        return href;
+    }
+    
+    public String formatHtmlLink(String link) {
+        return link.replaceAll("&", "&amp;");
+    }
+
+    private MediaType getMediaType(Resource resource) {
+        if (resource.getType() == Type.DIRECTORY) {
+            return getFormatGet(false).getMediaType();
+        } else if (resource.getType() == Type.RESOURCE) {
+            String mimeType = URLConnection.guessContentTypeFromName(resource.name());
+            if (mimeType == null || MediaType.APPLICATION_OCTET_STREAM.getName().equals(mimeType)) {
+                //try guessing from data
+                try (InputStream is = new BufferedInputStream(resource.in())) {
+                    mimeType = URLConnection.guessContentTypeFromStream(is);
+                } catch (IOException e) {
+                    //do nothing, we'll just use application/octet-stream
+                }
+            }
+            return mimeType == null ? MediaType.APPLICATION_OCTET_STREAM : MediaType.valueOf(mimeType);
+        } else {
+            return null;
+        }
+    }
+    
+    @Override
+    protected List<DataFormat> createSupportedFormats(Request request, Response response) {
+        List<DataFormat> formats = new ArrayList<DataFormat>();
+        
+        //HTML
+        if (operation == Operation.METADATA || resource.getType() == Type.DIRECTORY) {
+            formats.add(new ReflectiveHTMLFormat(request,response,this) {   
+                @Override
+                protected Configuration createConfiguration(Object data, Class clazz) {
+                    Configuration cfg = super.createConfiguration(data, clazz);
+                    cfg.setClassForTemplateLoading(ResourceResource.class, "");
+                    cfg.setObjectWrapper(new ObjectToMapWrapper<ResourceMetadata>(ResourceMetadata.class) {
+                        @Override
+                        protected void wrapInternal(Map properties, SimpleHash model, ResourceMetadata object) {
+                            super.wrapInternal(properties, model, object);
+                            properties.put("path", object.getParent() == null ? "/" :
+                                Paths.get(object.getParent().getPath(), object.getName()) ); 
+                            properties.put("parent_path", object.getParent() == null ? "" : object.getParent().getPath());
```

Comment:
@NielsCharlier the best thing about Eclipse formatting is that you can turn it off and on wherever you choose. Our published formatter enables the default off/on tags (in the Code Style / Formatter preference hit Edit and look for the Off/On Tags tab):
```
//@formatter:off
System
  .out
    .println(
"Eclipse will not format this"
    );
//@formatter:on
```
The purpose of the formatter and line length limits is to enhance readability. Eclipse often wraps lines in ways that reduce readability; a well-placed trailing ```//``` or formatter off/on tags is an acceptable solution where you feel this is justified.